job-server/src/spark.jobserver/JobManagerActor.scala

```diff
@@ -320,10 +320,12 @@ class JobManagerActor(contextConfig: Config) extends InstrumentedActor {
         case e: java.lang.AbstractMethodError => {
           logger.error("Oops, there's an AbstractMethodError... maybe you compiled " +
             "your code with an older version of SJS? here's the exception:", e)
+          statusActor ! JobErroredOut(jobId, DateTime.now(), e)
```

Comment:
@hbhanawat @soubhik-c  After thorough debugging found that Future was not getting stuck only in case of few types of error like NoClassDefFoundError and if we warp this error to some other exception then it was properly propagated. After debugging found that only [NonFatal](https://github.com/scala/scala/blob/2.12.x/src/library/scala/util/control/NonFatal.scala) exceptions are propagated from Future.It is also mentioned in [SIP-14](http://docs.scala-lang.org/sips/completed/futures-promises.html#exceptions) and there is a jira for that [SCALA-8938] https://issues.scala-lang.org/browse/SI-8938.